central/sensor/service/connection/messagestream/sizing_events.go

```diff
@@ -0,0 +1,27 @@
+package messagestream
+
+import (
+	"math"
+
+	"github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/reflectutils"
+)
+
+type sizingEventStream struct {
+	stream  CentralMessageStream
+	maxSeen map[string]float64
+}
+
+func (s *sizingEventStream) Send(msg *central.MsgToSensor) error {
+	typ := reflectutils.Type(msg)
+	gaugeValue := math.Max(s.maxSeen[typ], float64(msg.Size()))
+	metrics.SetGRPCMaxMessageSizeGauge(typ, gaugeValue)
+	s.maxSeen[typ] = gaugeValue
```

Comment:
[nit] you could avoid using `gaugeValue`
```suggestion
	s.maxSeen[typ] := math.Max(s.maxSeen[typ], float64(msg.Size()))
	metrics.SetGRPCMaxMessageSizeGauge(typ, s.maxSeen[typ])
```

Reply:
It was like that
<img width="815" alt="image" src="https://github.com/stackrox/stackrox/assets/6811076/da941cd1-dafc-4a03-a336-483569cc0a12">